{"pageProps":{"questions":[{"id":"xaxQ3DTjpoWjFpaMgXh9","exam_id":194,"question_id":216,"answer":"C","answer_description":"","answer_ET":"C","question_text":"What is the expected output of the following code?\n//IMG//","isMC":true,"answers_community":["C (100%)"],"topic":"4","timestamp":"2025-03-25 10:00:00","question_images":["https://img.examtopics.com/pcep-30-02/image363.png"],"discussion":[{"poster":"Donny_575","timestamp":"1742893200.0","upvote_count":"1","comment_id":"1409965","content":"Selected Answer: C\nLine 1: x = 42\n A global variable x is set to 42.\nLine 4–8: Function Definition\n def func():\n global x # Refers to the global 'x' global x tells Python: “When I refer to x in here, I mean the one outside the function.”\n print('1. x:', x)\n x = 23 # Modifies the global variable\n print('2. x:', x)\nLine 11: func() #Calls func():\nLine 6: #prints 1. x: 42\nLine 7: #sets x = 23 (updates the global x)\nLine 8: # prints 2. x: 23\nLine 12: print('3. x:', x) #Prints the updated value of global x, which is now 23\nFinal Output:\n1. x: 42\n2. x: 23\n3. x: 23"}],"unix_timestamp":1742893200,"answer_images":[],"url":"https://www.examtopics.com/discussions/python-institute/view/169813-exam-pcep-30-02-topic-4-question-28-discussion/","choices":{"A":"1 | 1. x: 42\n2 | 2. x: 42\n3 | 3. x: 42","C":"1 | 1. x: 42\n2 | 2. x: 23\n3 | 3. x: 23","D":"1 | 1. x: 42\n2 | 2. x: 23\n3 | 3. x: 42","B":"None of the above."}},{"id":"ysqBj7vjWIdGIwWcTYuD","isMC":true,"question_images":["https://img.examtopics.com/pcep-30-02/image365.png"],"url":"https://www.examtopics.com/discussions/python-institute/view/169474-exam-pcep-30-02-topic-4-question-29-discussion/","answer_images":[],"topic":"4","unix_timestamp":1742470500,"timestamp":"2025-03-20 12:35:00","discussion":[{"timestamp":"1742490420.0","poster":"akumo","comment_id":"1401209","content":"Selected Answer: D\nD. The code is erroneous.","upvote_count":"2"},{"comment_id":"1401010","upvote_count":"2","timestamp":"1742470500.0","poster":"Oluwafemi_Akinlosotu","content":"Selected Answer: D\nThe code is erroneous because print(func(func(2)) + 1) will first become print(func(1) + 1), which will then become print(None + 1), and this will throw a TypeError because arithmetic operations are impossible between the NoneType and int. \n\nN.B. True and False have values of 1 and 0 respectively, and those are of the boolean type (parallel scenario)."}],"question_id":217,"answers_community":["D (100%)"],"answer_ET":"D","choices":{"D":"The code is erroneous.","C":"2","A":"None","B":"1"},"question_text":"What is the expected output of the following code?\n//IMG//","answer":"D","exam_id":194,"answer_description":""},{"id":"yabzCYisaWzRxEgfOydi","answers_community":["B (100%)"],"discussion":[{"content":"Selected Answer: B\nB. def","poster":"akumo","upvote_count":"1","comment_id":"1401171","timestamp":"1742486280.0"}],"answer_images":[],"topic":"4","timestamp":"2025-03-20 16:58:00","unix_timestamp":1742486280,"choices":{"A":"fun","C":"function","B":"def"},"exam_id":194,"answer_description":"","answer":"B","question_id":218,"question_images":[],"url":"https://www.examtopics.com/discussions/python-institute/view/169525-exam-pcep-30-02-topic-4-question-3-discussion/","question_text":"A function definition starts with the keyword:","answer_ET":"B","isMC":true},{"id":"ZxHij0c0Idiv5b6iRbKP","timestamp":"2025-03-25 10:14:00","answer_images":[],"topic":"4","question_text":"What is the expected output of the following code?\n//IMG//","unix_timestamp":1742894040,"answer":"C","answer_ET":"C","isMC":true,"discussion":[{"poster":"Donny_575","content":"Selected Answer: C\nC. yield accumulates the value of s throughout the 3 interations * + ** + ***, with an output of ******","timestamp":"1742894040.0","upvote_count":"1","comment_id":"1409972"}],"url":"https://www.examtopics.com/discussions/python-institute/view/169814-exam-pcep-30-02-topic-4-question-30-discussion/","exam_id":194,"answer_description":"","question_images":["https://img.examtopics.com/pcep-30-02/image367.png"],"choices":{"D":"***","A":"*","B":"The code is erroneous.","C":"******"},"answers_community":["C (100%)"],"question_id":219},{"id":"s7Ps0W3Ed2mqN6GxoAtJ","exam_id":194,"discussion":[{"timestamp":"1742490600.0","content":"Selected Answer: A\nA. its position within the argument list","comment_id":"1401213","upvote_count":"1","poster":"akumo"}],"isMC":true,"timestamp":"2025-03-20 18:10:00","choices":{"C":"its connection with existing variables","A":"its position within the argument list","D":"the argument’s name specified along with its value","B":"its value"},"question_id":220,"question_text":"The meaning of a positional argument is determined by:","answer_images":[],"unix_timestamp":1742490600,"answers_community":["A (100%)"],"answer":"A","question_images":[],"topic":"4","answer_ET":"A","answer_description":"","url":"https://www.examtopics.com/discussions/python-institute/view/169546-exam-pcep-30-02-topic-4-question-31-discussion/"}],"exam":{"lastUpdated":"12 Apr 2025","name":"PCEP-30-02","isBeta":false,"isMCOnly":true,"id":194,"numberOfQuestions":330,"provider":"Python-institute","isImplemented":true},"currentPage":44},"__N_SSP":true}